src/plugins/data/public/ui/query_editor/language_switcher.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiComboBox, EuiComboBoxOptionOption, PopoverAnchorPosition } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import React from 'react';
+import { getSearchService, getUiService } from '../../services';
+
+interface Props {
+  language: string;
+  onSelectLanguage: (newLanguage: string) => void;
+  anchorPosition?: PopoverAnchorPosition;
+  appName?: string;
+}
+
+function mapExternalLanguageToOptions(language: string) {
+  return {
+    label: language,
+    value: language,
+  };
+}
+
+export function QueryLanguageSwitcher(props: Props) {
```

Comment:
there is already a query language selector, how is this different?

Reply:
this is not added by me but seems like appeared after the rebase..will check